### ISSUE TITLE
ai-revision does nothing when specifying a file

### ISSUE THREAD

**Issue by SilasK** [manubot/manubot-ai-editor#17]
I run the ai-revision.
Without specifying any file it only updates the introduction.

If I specify a file it does aparently nothing and the pull request fails
https://github.com/SilasK/Research_Proposal/actions/runs/5134823028/jobs/9239373226

**Comment by miltondp** [comment]
Hi @SilasK, thank you for your interest. The tool uses section-specific prompts to revise the text, and it will try to infer the section from the file names. If this fails, which seems to be your case, you need to indicate which section each file belongs to. I've added this documentation in [this PR](https://github.com/manubot/rootstock/pull/496/files), and it would be great if you could go through it and let us know if it is clear and help you solve the problem.

**Comment by SilasK** [comment]
Thank you for your reply.

Sorry, yes it's private.

Nothing special in the log:
``` 
Run manubot ai-revision --content-directory content/
  manubot ai-revision --content-directory content/
  shell: /usr/bin/bash --login {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.3/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.3/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.3/x64
    Python[2](https://github.com/SilasK/Research_Proposal/actions/runs/5134892888/jobs/9239537171#step:5:2)_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/SilasK/Research_Proposal/actions/runs/5134892888/jobs/9239537171#step:5:3).11.3/x6[4](https://github.com/SilasK/Research_Proposal/actions/runs/5134892888/jobs/9239537171#step:5:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.3/x[6](https://github.com/SilasK/Research_Proposal/actions/runs/5134892888/jobs/9239537171#step:5:6)4
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.[11](https://github.com/SilasK/Research_Proposal/actions/runs/5134892888/jobs/9239537171#step:5:11).3/x64/lib
    OPENAI_API_KEY: ***
    AI_EDITOR_LANGUAGE_MODEL: text-davinci-003
    AI_EDITOR_FILENAMES_TO_REVISE: 03.aim_subspecies_quant.md
Using language model from environment variable 'AI_EDITOR_LANGUAGE_MODEL'
Language model:  text-davinci-003
File names to revise: {'03.aim_subspecies_quant.md'}

```
[raw_log_ai_revision.txt](https://github.com/manubot/manubot-ai-editor/files/11618528/raw_log_ai_revision.txt)

**Comment by SilasK** [comment]
Ok, I will add the file mapping. 

My manuscript is actualy more a grant proposal. than an article.
I asume in the mapping I can give arbitrary description, can I?

Is there a simple way to modify the query context sent to the AI model?

**Comment by miltondp** [comment]
Yes, you can use whatever you want as the section name. In that case, it will default to a general prompt, as you can see [here](https://github.com/manubot/manubot-ai-editor/blob/a571f4dde356a212d3e37fc0d6c5a8ff67e5b935/libs/manubot_ai_editor/models.py#L325). Keep in mind also that that general prompt mentions "... of an academic paper...", which might or might not affect the text revision of a grant proposal.

Regarding custom prompts, I agree that it will be better in your case, and we will be working on that soon. However, there is a work-in-progress feature that you can use: add this environment variable in the `.github/workflows/ai-revision.yaml` file (in the "Revise manuscript" step) to provide whatever prompt you want:

```
AI_EDITOR_CUSTOM_PROMPT: "Proofread the following paragraph"
```

Again, if you give it a try, please give us feedback on whether it worked as expected or not.

**Comment by SilasK** [comment]
I made it work for both the file specification and the custom prompt. 
I don't know if it is possible to use the file specification in the custom prompt as in the original one?

**Comment by miltondp** [comment]
Thanks for the feedback!

You can't use a custom prompt per file if that's what you mean. But sounds like a good use case. We'll be working on these use cases soon, so if you can provide a more detailed specification of what you would like to have, that would be really helpful.
